src/raft.rs

```diff
@@ -2457,4 +2477,73 @@ impl<T: Storage> Raft<T> {
         to_send.set_entries(req.take_entries());
         Some(to_send)
     }
+
+    /// Reduce size of 'ents' from uncommitted size.
+    ///
+    /// # Panics
+    ///
+    /// Panics if size of 'ents' is greater than uncommitted size
+    pub fn reduce_uncommitted_size(&mut self, ents: &[Entry]) {
+        // fast path for NO_LIMIT and non-leader endpoint
+        if self.state != StateRole::Leader || self.max_uncommitted_size == NO_LIMIT as usize {
+            return;
+        }
+
+        let mut size: usize = 0;
+        for entry in ents {
+            size += entry.get_data().len()
+        }
+
+        if size > self.uncommitted_size {
+            fatal!(
+                self.r.logger,
+                "try to reduce uncommitted size less than 0, first index of pending ents is {}",
+                ents[0].get_index()
+            );
```

Comment:
How about set `self.uncommitted_size` to zero and print a warning log in this case like [etcd](https://github.com/etcd-io/etcd/blob/cc2b4cd05e6b677c9c73d9ccb7333ebf1e1be7ee/raft/raft.go#L1666-L1670)